src/app/core/services/issue.service.ts

```diff
@@ -85,24 +85,22 @@ export class IssueService {
     );
   }
 
-  /**
-   * This function will update the issue's state of the application. This function needs to be called whenever a issue is deleted.
-   */
-  deleteFromLocalStore(issueToDelete: Issue) {
-    const { [issueToDelete.id]: issueToRemove, ...withoutIssueToRemove } = this.issues;
-    this.issues = withoutIssueToRemove;
-    this.issues$.next(Object.values(this.issues));
-  }
-
   /**
    * This function will update the issue's state of the application. This function needs to be called whenever a issue is added/updated.
+   *
+   * @params issuesToUpdate - An array of issues to update the state of the application with.
+   * @params shouldEmit - Whether the updated issues should be emitted to issues$.
    */
-  updateLocalStore(issueToUpdate: Issue) {
-    this.issues = {
-      ...this.issues,
-      [issueToUpdate.id]: issueToUpdate
-    };
-    this.issues$.next(Object.values(this.issues));
+  updateLocalStore(issuesToUpdate: Issue[], shouldEmit: boolean = true) {
```

Comment:
Since `shouldEmit` is true by default, and all methods which are calling `createAndSaveIssueModels` (`initializeData` and `getLatestIssue`) do not pass the `shouldEmit` parameter, it seems that we do not need the `shouldEmit` parameter in the first place. Besides, if there are any updates to the state, changes should be made known to the rest of the application; so let's look at removing the shouldEmit parameter for now until we find a use case for needing to selectively emit issues which are updated.

Reply:
I think the original idea was that we can have some sort of function that has this kind of a calling behaviour:

```
shouldEmit = false
shouldEmit = false
...
shouldEmit = true  # at the end
```

I've removed it, I agree that it can be added back if this behaviour is needed in the future!